server/text_generation_server/models/custom_modeling/flash_santacoder_modeling.py

```diff
@@ -179,22 +181,40 @@ def load_col(config, prefix: str, weights, bias: bool):
         bias = weights.get_sharded(f"{prefix}.bias", dim=0)
     else:
         bias = None
-    return TensorParallelColumnLinear(get_linear(weight, bias, config.quantize))
+    return TensorParallelColumnLinear(get_linear(weight, bias, config.quantize, device=weights.device))
 
 
 def load_row(config, prefix: str, weights, bias: bool):
+    quantize = config.quantize
+    if quantize == "gptq-cuda" and weights.process_group.size() > 1:
+        g_idx = weights.get_tensor(f"{prefix}.g_idx")
+        groupsize = weights.get_tensor("gptq_groupsize").item()
+        
+        act_order = True
+        if g_idx is not None:
+            if torch.equal(g_idx.cpu(), torch.tensor([i // groupsize for i in range(g_idx.shape[0])], dtype=torch.int32)) or (g_idx == 0).all():
+                act_order = False
+            else:
+                # Exllama implementation does not support row tensor parallelism with act-order, as
+                # it would require to reorder input activations that are split unto several GPUs
+                quantize = "gptq"
+
     if config.transpose:
         weight = weights.get_sharded(f"{prefix}.weight", dim=0).T
     else:
-        weight = weights.get_multi_weights_row(prefix, quantize=config.quantize)
+        weight = weights.get_multi_weights_row(prefix, quantize=quantize)
 
     if bias and weights.process_group.rank() == 0:
         # Rank is only on the first rank process
         bias = weights.get_tensor(f"{prefix}.bias")
     else:
         bias = None
+    
+    if quantize == "gptq-cuda" and not act_order:
+        weight[3] = None  # remove g_idx to indicate to exllama that act-order is not used
```

Comment:
Just don't modify anything here, let this flow to `get_linear` which can do;

```python
if torch.equal(g_idx.cpu(), torch.tensor([i // groupsize for i in range(g_idx.shape[0])], dtype=torch.int32)) or (g_idx == 0).all():
```

and decide there if we should use the triton or the cuda kernel.